src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/CodeownersGenerator.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public interface ICodeownersGenerator
+{
+    Task UpdateCodeowners(CancellationToken cancellationToken);
+}
+
+public class CodeownersGenerator : ICodeownersGenerator
+{
+    private const string CodeownersHeader = "### CONTENT BELOW IS AUTO-GENERATED AND MANUAL CHANGES WILL BE OVERWRITTEN ###";
+
+    private static readonly IReadOnlyCollection<LocalPath> s_codeownersLocations = new[]
+    {
+        new UnixPath(VmrInfo.CodeownersFileName),
+        new UnixPath(".github/" + VmrInfo.CodeownersFileName),
```

Comment:
While this is rare, not every repository has the `CODEOWNERS` file under `./github`. Specifically, [`installer`](https://github.com/dotnet/installer/blob/main/CODEOWNERS) and [`vstest`](https://github.com/microsoft/vstest/blob/main/CODEOWNERS) have their `CODEOWNERS` in the root of the repo